From: Vincent Bernat <vincent@bernat.im>
Date: Sun, 8 Oct 2017 19:14:56 +0200
Subject: configuration: correctly write ini/environment for tcp.bind

In 3791aee46667, tcp.bind was changed from an IP to a list. However,
when using `--fi` or `--fe`, the list was written as a
list (`[1.2.3.4]`) instead of a space separated list of IP.
---
 lib/exabgp/configuration/environment.py | 4 ++++
 lib/exabgp/configuration/setup.py       | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/lib/exabgp/configuration/environment.py b/lib/exabgp/configuration/environment.py
index 12ef666..95c7676 100644
--- a/lib/exabgp/configuration/environment.py
+++ b/lib/exabgp/configuration/environment.py
@@ -103,6 +103,10 @@ class environment (object):
 	def quote (_):
 		return "'%s'" % str(_)
 
+	@staticmethod
+	def quote_list (_):
+		return "'%s'" % ' '.join([str(x) for x in _])
+
 	@staticmethod
 	def nop (_):
 		return _
diff --git a/lib/exabgp/configuration/setup.py b/lib/exabgp/configuration/setup.py
index b38a663..d635a3b 100644
--- a/lib/exabgp/configuration/setup.py
+++ b/lib/exabgp/configuration/setup.py
@@ -191,7 +191,7 @@ environment.configuration = {
 		},
 		'bind': {
 			'read':  environment.ip_list,
-			'write': environment.quote,
+			'write': environment.quote_list,
 			'value': '',
 			'help':  'Space separated list of IPs to bind on when listening (no ip to disable)',
 		},
